### ISSUE TITLE
superfluous-parens in if-expression are not superfluous

### ISSUE THREAD

**Issue by andreasvc** [PyCQA/pylint#2818]
### Steps to reproduce
Create file t.py:
```
def f(b):
    return [a for a in (b if b else [b])]
```
Related to #574 but this example has an if-expression instead of short circut.

### Current behavior
```
t.py:2:0: C0325: Unnecessary parens after 'in' keyword (superfluous-parens)
```
Note that following this suggestion actually creates a SyntaxError:
```
def g(b):
    return [a for a in b if b else [b]]
```
```
t.py:2:0: E0001: invalid syntax (<unknown>, line 2) (syntax-error)
```
### Expected behavior
This warning is not emitted in this case.

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.2+ (default, Feb 27 2019, 15:41:59) 
[GCC 8.2.0]
```

**Comment by PCManticore** [comment]
Thanks @andreasvc for reporting this issue.

**Comment by p-l-** [comment]
I'm also having this bug in an expression that matches @andreasvc's pattern:
 ```Python
    rec["answers"] = [elt.lower() for elt in (rec["answers"] if rec["answers"] else [])]
```
